### PR TITLE
Week1-Assignment

### PR DESCRIPTION
**Pull request by Radhikajram**

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -30,13 +30,149 @@
...
+    loadDiv(userRepo, selectedValue);
+    selectBox.onchange = function() {
+      loadDiv(userRepo, selectBox.value);
+    };
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
+      const myParent = document.body;
```
Comment (grod220) [inline homework/index.js]: Who is my? I'd call it something like `BodyEl`

File: homework/index.js [hunk 2]
```diff
@@ -30,13 +30,149 @@
...
+
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        // Create div element 'select' in  document body to hold the label element and list box.
+
+        createAndAppend('div', myParent, { id: 'select' });
```
Comment (grod220) [inline homework/index.js]: The id name is a bit confusing here. Would call is something like `select-container` or something.

File: homework/index.js [hunk 3]
```diff
@@ -30,13 +30,149 @@
...
+          id: 'label',
+          for: 'repo',
+        });
+        createAndAppend('select', select, { id: 'myselect' });
+
+        // Create two div elements section1 and section2 under 'Root' div to have
+        // section1 - Repository Information.
+        // section2 - Contributions.
+
+        createAndAppend('div', root, { id: 'section1' });
```
Comment (grod220) [inline homework/index.js]: Would name these divs closer to what content they will hold. More like `contributors` and `repo-details`

File: homework/index.js [hunk 4]
```diff
@@ -30,13 +30,149 @@
...
+
+        // Insert section1 before section2 div element under 'root' div.
+        const newNode = document.getElementById('section2');
+        const referenceNode = document.querySelector('section1');
+        root.insertBefore(newNode, referenceNode);
+
+        // Insert Select div first in the body before root div.
+        myParent.insertBefore(select, document.getElementById('root'));
+
+        loadSelectionValues(JSON.stringify(data, null, 2));
```
Comment (grod220) [inline homework/index.js]: Does it need to be stringified? Kinda confused what the other two parameters do.

File: homework/index.js [hunk 5]
```diff
@@ -30,13 +30,149 @@
...
+    document.getElementById('section1').innerHTML = htmlString;
+  }
+
+  // Create options under 'SELECT' element which should have HYF Repositories.
+
+  function loadSelectionValues(userInfo) {
+    const sortRepoName = [];
+
+    const userRepo = JSON.parse(userInfo);
+    const mySelect = document.getElementById('myselect');
```
Comment (grod220) [inline homework/index.js]: Not sure what the word `my` helps signify. Think a better name could be used?

File: homework/index.js [hunk 6]
```diff
@@ -30,13 +30,149 @@
...
+
+    // push all the HYP repo names to sort array.
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const repo in userRepo) {
```
Comment (grod220) [inline homework/index.js:111]: arrays should use for...of loops

File: homework/index.js [hunk 7]
```diff
@@ -30,13 +30,149 @@
...
+
+  function loadSelectionValues(userInfo) {
+    const sortRepoName = [];
+
+    const userRepo = JSON.parse(userInfo);
+    const mySelect = document.getElementById('myselect');
+
+    // push all the HYP repo names to sort array.
+
+    // eslint-disable-next-line no-restricted-syntax
```
Comment (grod220) [inline homework/index.js:110]: Let's remove these eslint-disable comments in the PR

File: homework/index.js (hunk 6 above)
Comment (grod220) [reply homework/index.js:111]: that way you won't need to do `hasOwnProperty` below

File: homework/index.js [hunk 8]
```diff
@@ -30,13 +30,149 @@
...
+    for (const repo in userRepo) {
+      if (Object.prototype.hasOwnProperty.call(userRepo, repo)) {
+        sortRepoName.push(userRepo[repo].name);
+      }
+    }
+
+    // sort the repo name using sort function and localeComapare for uppercase and lowercase sorting.
+    sortRepoName.sort((a, b) => a.localeCompare(b));
```
Comment (grod220) [inline homework/index.js:118]: Nice, I've never heard of `localCompare` before

File: homework/index.js [hunk 9]
```diff
@@ -30,13 +30,149 @@
...
+      option.value = repo;
+      option.text = repo;
+      mySelect.appendChild(option);
+    }
+
+    const selectBox = document.getElementById('myselect');
+    const selectedValue = selectBox.options[selectBox.selectedIndex].value;
+
+    // Load Repository information for the choose repository name in the select box.
+    loadDiv(userRepo, selectedValue);
```
Comment (grod220) [inline homework/index.js]: Great job creating separate functions. But can we rename this to something more clear on what it does?

File: homework/index.js [hunk 10]
```diff
@@ -30,13 +30,149 @@
...
+            '<div id="row">',
+            `${'<p id="name-info">'} Repository name : ${'<a href="'}${
+              repoInfo[repo].html_url
+            }"${'/>'} ${repoInfo[repo].name}</a></p>`,
+            `${'<p id="desc">'}  Description :  ${repoInfo[repo].description}</p>`,
+            `${'<p id="forks">'}  Forks       : ${repoInfo[repo].forks_count}</p>`,
+
+            `${'<p id="updated">'} Updated  : ${formatedUpdate}</p>`,
+            '</div>',
+          );
```
Comment (grod220) [inline homework/index.js:82]: Interesting technique here. Yes, this will work.

File: homework/index.js [hunk 11]
```diff
@@ -30,13 +30,149 @@
...
+
+          // Call another function(addContributors) to fill i contributor information.
+          fetchJSON(repoInfo[repo].contributors_url, (err, data) => {
+            const root = document.getElementById('root');
+
+            if (err) {
+              createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+            } else {
+              addContributors(data);
```
Comment (grod220) [inline homework/index.js:91]: Nice making a new function here 👍

File: package.json [hunk 12]
```diff
@@ -9,7 +9,7 @@
   "author": "Jim Cramer",
   "license": "CC-BY-4.0",
   "devDependencies": {
-    "eslint": "^5.9.0",
+    "eslint": "^5.16.0",
```
Comment (grod220) [inline package.json:12]: Necessary?

**Review by grod220 — CHANGES_REQUESTED** [review]
Great job!

File: homework/index.js (hunk 1 above)
Comment (Radhikajram) [reply homework/index.js]: changed myParent to BodyEl.

File: homework/index.js (hunk 2 above)
Comment (Radhikajram) [reply homework/index.js]: Corrected the container name for select as - selectcontainer.

File: homework/index.js (hunk 3 above)
Comment (Radhikajram) [reply homework/index.js]: corrected the name div name as repodetails,contributorinfomation

File: homework/index.js (hunk 4 above)
Comment (Radhikajram) [reply homework/index.js]: It was there in existing code, but i corrected ans used the data as such.

File: homework/index.js (hunk 5 above)
Comment (Radhikajram) [reply homework/index.js]: corrected with name selectrepo.

File: homework/index.js (hunk 9 above)
Comment (Radhikajram) [reply homework/index.js]: Renamed the function name as loadRepoDetails.

**Review by grod220 — APPROVED** [review]
Going to approve the week 1 stuff and request changes for week 2

**Pull request merged**